js/bybit.js

```diff
@@ -1947,142 +1866,103 @@ module.exports = class bybit extends Exchange {
-                request['endTime'] = this.sum (since, limit * duration * 1000);
-            }
-        }
-        if (limit !== undefined) {
-            request['limit'] = limit; // max 1000, default 1000
-        }
-        request['interval'] = timeframe;
-        const response = await this.publicGetSpotV3PublicQuoteKline (this.extend (request, params));
-        //
-        //     {
-        //         "retCode": 0,
-        //         "retMsg": "OK",
-        //         "result": {
-        //         "list": [
-        //             {
-        //             "t": 1659430380000,
-        //             "s": "BTCUSDT",
-        //             "sn": "BTCUSDT",
-        //             "c": "21170.14",
-        //             "h": "21170.14",
-        //             "l": "21127.86",
-        //             "o": "21127.86",
-        //             "v": "0.907276"
-        //             }
-        //         ]
-        //         },
-        //         "retExtInfo": {},
-        //         "time": 1659430400353
-        //     }
-        //
-        const result = this.safeValue (response, 'result', {});
-        const ohlcvs = this.safeValue (result, 'list', []);
-        return this.parseOHLCVs (ohlcvs, market, timeframe, since, limit);
-    }
-
-    async fetchDerivativesOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name bybit#fetchOHLCV
+         * @description fetches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the bybit api endpoint
+         * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        this.checkRequiredSymbol ('fetchOHLCV', symbol);
         await this.loadMarkets ();
         const market = this.market (symbol);
-        if (market['option']) {
-            throw new NotSupported (this.id + ' fetchOHLCV() is not supported for option markets');
-        }
         const request = {
             'symbol': market['id'],
         };
         const duration = this.parseTimeframe (timeframe);
-        const now = this.milliseconds ();
         if (limit === undefined) {
             limit = 200; // default is 200 when requested with `since`
-        } else {
-            request['limit'] = limit;
         }
         if (since === undefined) {
-            since = now - (limit * duration * 1000);
+            request['start'] = since;
```

Comment:
This is surely wrong.

Running this branch with "None" as since yields an exception.

``` python
import ccxt
ct = ccxt.bybit({'options': {'defaultType': 'swap'}})
ct.fetch_ohlcv('BTC/USDT:USDT', timeframe='1m', since=None, limit=200 )
```

It's however a perfectly valid call on the latest master version.

```
fetch Request: bybit GET https://api.bybit.com/v5/market/kline?symbol=BTCUSDT&start=None&limit=200&interval=60&category=linear RequestHeader
s: {'User-Agent': 'python-requests/2.28.2', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: None                        
                                                                      
fetch Response: bybit GET https://api.bybit.com/v5/market/kline?symbol=BTCUSDT&start=None&limit=200&interval=60&category=linear 200 Response
Headers: {'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '102', 'Ret_code': '10001', 'Traceid': '1ea9ee8a78a0ddcd6dc6e
b43f41e9cc1', 'Timenow': '1676654468321', 'Server': 'Openresty', 'Expires': 'Fri, 17 Feb 2023 17:21:08 GMT', 'Cache-Control': 'max-age=0, no
-cache, no-store', 'Pragma': 'no-cache', 'Date': 'Fri, 17 Feb 2023 17:21:08 GMT', 'Connection': 'keep-alive'} ResponseBody: {"retCode":10001
,"retMsg":"Request parameter error.","result":{},"retExtInfo":{},"time":1676654468321}                                                
---------------------------------------------------------------------------
BadRequest                                Traceback (most recent call last)
Cell In [3], line 1                                                   
----> 1 ct.fetch_ohlcv('BTC/USDT:USDT', timeframe='1m', since=None, limit=200 )
                                                                                                                                            
File ~/devel/cryptos/ccxt/python/ccxt/bybit.py:1877, in bybit.fetch_ohlcv(self, symbol, timeframe, since, limit, params)
   1875     else:
   1876         raise NotSupported(self.id + ' fetchOHLCV() is not supported for option markets')
-> 1877 response = getattr(self, method)(self.extend(request, params))

```

Reply:
@sc0Vu I think you meant `!== undefined` instead

Reply:
@carlosmiei @sc0Vu i guess this one was forgotten?

Reply:
@xmatthias u're right 🤦 , going to patch it asap